core/src/main/java/org/apache/accumulo/core/clientImpl/lexicoder/ByteUtils.java

```diff
@@ -76,6 +76,10 @@
     for (int i = 0; i < in.length; i++) {
       if (in[i] == 0x01) {
         i++;
+        if (i >= in.length) {
+          throw new IllegalArgumentException("Bad bytes when attempting to unescape. Expected "
+              + "more bytes after last byte " + in[in.length - 1]);
```

Comment:
I think you could do `String.format` to print the byte as hex here.

Reply:
What are you thinking, like this? `String.format("x%02x", in[in.length - 1]))`

Reply:
Yeah, pretty much that. Just to help people think of it as a byte, rather than some random number.